### ISSUE TITLE
Crash in Document.FindDocument

### ISSUE THREAD

**Issue by sharwell** [kaby76/AntlrVSIX#23]
🔗 Originally reported here: https://developercommunity.visualstudio.com/content/problem/796874/intellisense-crashes-visual-studio-2019.html

```
Event Time (UTC): 10/28/2019 7:25:24 PM
Event ID: 1026
Data: Application: devenv.exe
Framework Version: v4.0.30319
Description: The process was terminated due to an unhandled exception.
Exception Info: System.NullReferenceException
   at Workspaces.Document.FindDocument(System.String)
   at Workspaces.Project.FindDocument(System.String)
   at Workspaces.Workspace.FindDocument(System.String)
   at AntlrVSIX.AggregateTagger.AntlrClassifier.ReparseFile(System.Object, Microsoft.VisualStudio.Text.TextContentChangedEventArgs)
   at AntlrVSIX.AggregateTagger.AntlrClassifier+<OnTextChanged>d__13.MoveNext()
   at System.Runtime.CompilerServices.AsyncMethodBuilderCore+<>c.<ThrowAsync>b__6_0(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.DispatcherOperation.InvokeImpl()
   at System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(System.Object)
   at MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(System.Object)
   at System.Threading.ExecutionContext.RunInternal(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object, Boolean)
   at System.Threading.ExecutionContext.Run(System.Threading.ExecutionContext, System.Threading.ContextCallback, System.Object)
   at MS.Internal.CulturePreservingExecutionContext.Run(MS.Internal.CulturePreservingExecutionContext, System.Threading.ContextCallback, System.Object)
   at System.Windows.Threading.DispatcherOperation.Invoke()
   at System.Windows.Threading.Dispatcher.ProcessQueue()
   at System.Windows.Threading.Dispatcher.WndProcHook(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndWrapper.WndProc(IntPtr, Int32, IntPtr, IntPtr, Boolean ByRef)
   at MS.Win32.HwndSubclass.DispatcherCallbackOperation(System.Object)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(System.Delegate, System.Object, Int32)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(System.Object, System.Delegate, System.Object, Int32, System.Delegate)
   at System.Windows.Threading.Dispatcher.LegacyInvokeImpl(System.Windows.Threading.DispatcherPriority, System.TimeSpan, System.Delegate, System.Object, Int32)
   at MS.Win32.HwndSubclass.SubclassWndProc(IntPtr, Int32, IntPtr, IntPtr)
```

**Comment by kaby76** [comment]
Of course, a file name is sometimes not available from an ITextBuffer. Checked in fixes to no-op taggers and other code. Thanks.

**Comment by kaby76** [comment]
And, yes, it's probably not a good idea to throw exceptions from OnTextChanged if VS is going to crash. Will need to log these errors instead.

**Comment by kaby76** [comment]
Fixed with next release v4.0.3.
